### PR TITLE
YUNIKORN-466: Node capacity update does not update root queue or partition

### PR DESCRIPTION
**Pull request by manirajv06**
Added updateNode in Partition object to update the partition resource and root queue resource whenever node resources changes.

### CODE REVIEW

File: pkg/scheduler/tests/operation_test.go [hunk 1]
```diff
@@ -378,6 +383,67 @@ partitions:
...
 	assert.Equal(t, int64(schedulingNode1.GetAllocatedResource().Resources[resources.MEMORY]), int64(0))
 	assert.Equal(t, int64(schedulingNode1.GetAvailableResource().Resources[resources.MEMORY]), int64(300))
+
+	newRes, err = resources.NewResourceFromConf(map[string]string{"memory": "300", "vcore": "10"})
+	assert.NilError(t, err, "failed to create resource")
+	assert.Equal(t, newRes.DAOString(), partitionInfo.GetTotalPartitionResource().DAOString())
+	assert.Equal(t, newRes.DAOString(), partitionInfo.GetQueue("root").GetMaxResource().DAOString())
+
+	// Register a node
+	err = ms.proxy.Update(&si.UpdateRequest{
```
Comment (wilfred-s) [inline pkg/scheduler/tests/operation_test.go:393]: We have a node already, we can perform the second part of the test (lowering the resources) using the same node.

**Review by wilfred-s — CHANGES_REQUESTED** [review]
Can you also add unit test for the newly created method as part of the partition tests?

File: pkg/scheduler/partition.go [hunk 2]
```diff
@@ -586,6 +586,16 @@ func (pc *PartitionContext) removeNode(nodeID string) []*objects.Allocation {
 	return pc.removeNodeInternal(nodeID)
 }
 
+// Update a node
```
[the remark this answers is not in the document]
Comment (manirajv06) [reply pkg/scheduler/partition.go:589]: > Can you also add unit test for the newly created method as part of the partition tests?

Added.

[the remark this answers is not in the document]
Comment (manirajv06) [reply pkg/scheduler/partition.go:589]: > Explain what you update on the node and why that triggers the partition update etc.

On receiving node update event signal, context#updateNodes calls partition#updateNode() to set the correct value for both partition total resource and root queue max resource.

File: pkg/scheduler/tests/operation_test.go [hunk 3]
```diff
@@ -371,13 +371,162 @@ partitions:
...
+					},
+				},
+				Action: si.UpdateNodeInfo_UPDATE,
+			},
+		},
+		RmID: "rm:123",
+	})
+	assert.NilError(t, err, "UpdateRequest failed")
+	waitForAvailableNodeResource(t, ms.scheduler.GetClusterContext(), "[rm:123]default",
+		[]string{"node-1:1234"}, 100, 1000)
```
Comment (wilfred-s) [inline pkg/scheduler/tests/operation_test.go:408]: No need to split the line to keep length down, make it one line. (multiple times found)

**Review by wilfred-s — CHANGES_REQUESTED** [review]
Please see the comments, change is looking good with some minor issues.

**Review by wilfred-s — APPROVED** [review]
conflicts fixed on commit

**Pull request closed without merging**